crates/bevy_winit/src/lib.rs

```diff
@@ -194,15 +199,23 @@ pub fn winit_runner(mut app: App) {
                 },
                 _ => {}
             },
-            event::Event::DeviceEvent { ref event, .. } => {
-                if let DeviceEvent::MouseMotion { delta } = event {
+            event::Event::DeviceEvent { ref event, .. } => match event {
+                DeviceEvent::MouseMotion { delta } => {
                     let mut mouse_motion_events =
                         app.resources.get_mut::<Events<MouseMotion>>().unwrap();
                     mouse_motion_events.send(MouseMotion {
                         delta: Vec2::new(delta.0 as f32, delta.1 as f32),
                     });
                 }
-            }
+                DeviceEvent::Motion { axis, value } => {
+                    let mut axis_events = app.resources.get_mut::<Events<Motion>>().unwrap();
```

Comment:
This leaks the internals of winit (via the winit axis id), which means users consuming this are taking a hard dependency on winit. I would prefer it if we provided our own abstraction here. Lets remove this for now and spec it out later.

Reply:
So do you mean something like bevy_device in the future? same as bevy_window, reimplementing winit.

Reply:
I think we would probably just extend bevy_input, but yeah it would be the same idea as bevy_window applied to controllers.